### DISCUSSION TITLE
Terraform lockhashes incorrectly cached

### DISCUSSION THREAD

**Discussion by Daniel-I-Am** [renovatebot/renovate#25601] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Docker 37.37.2-full

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

It seems in my recent PR (https://github.com/renovatebot/renovate/pull/25059) for introducing Terraform ziphashes, I introduced an issue that I was unable to catch during local testing, due to caching (seemingly) not being enabled in that environment. Today our self-hosted renovate made lockfile maintenance PRs that were not quite correct. Renovate had reused ziphashes for the wrong providers (it had reused the google-beta provider ziphashes for google (not-beta) provider). And in the PR linked in the reproduction repository (through mend), it even got incorrect ansible hashes from somewhere unknown.

----

I think this caching annotation is to blame. I can submit a PR to correct it, but want to make sure that this could be the cause.

https://github.com/renovatebot/renovate/blob/4798368f373947135ab9691f42b8efb759a04c60/lib/modules/datasource/terraform-provider/index.ts#L285-L290

----

reproduction:
- https://github.com/Daniel-I-Am/renovate-terraform-lock-zh-test/pull/7

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
What is wrong with the annotation?

**Reply by Daniel-I-Am** [reply]
It seems to be using variables (and types) that are not passed in to the method for the cache key. Therefore it could be caching the wrong things (especially since it is using an array turned into a string as key). So I think that's what is causing the behaviour I am seeing. Later in the code only `builds[0].shasums_url` is used, so I think that is what the cache key should be changed to in order to fix the issue I see.

**Reply by viceice** [reply]
can you fix it asap? otherwise we need to revert the zip hashes feature.

**Reply by Daniel-I-Am** [reply, marked as answer]
Understandable. Unfortunately, I was unable to take a look yesterday. I think https://github.com/renovatebot/renovate/pull/25626 should do the trick :crossed_fingers:
